ceph/rados/mgr_workflows.py

```diff
@@ -0,0 +1,274 @@
+        self.mon_obj = MonitorWorkflows(node=node)
+        self.cluster = node.cluster
+        self.config = node.config
+        self.client = node.cluster.get_nodes(role="client")[0]
+        self.cluster_nodes = node.cluster.get_nodes()
+
+    def add_mgr_service(self, **kwargs) -> bool:
+        """
+        Method to add the mgr service from the given host
+        Args:
+            kwargs: the accepted KW args for the method are:
+                host: Cluster object of the host from where mgr needs to be Added
+        returns:
+            Pass -> True, Fail -> false
+        """
+        # Checking if mgr service exists before removing
+        host = kwargs.get("host")
+        hostname = host.hostname
+
+        if self.check_mgr_exists_on_host(host=hostname):
+            log.info(
+                f"MGR daemon already present on the host {hostname}. Returning Pass"
+            )
+            return True
+
+        log.debug(f"Adding mgr daemon on the host : {hostname}")
+
+        # Adding lable to host
+        labels = self.mon_obj.get_host_labels(host=hostname)
+        if "mgr" not in labels:
+            cmd = f"ceph orch host label add {hostname} mgr"
+            self.client.exec_command(sudo=True, cmd=cmd)
+            time.sleep(10)
+            log.debug(f"Added mgr label for the host : {hostname}")
+
+        # Adding host as mgr
+        cmd = "ceph orch apply mgr --placement='label:mgr'"
+        self.client.exec_command(sudo=True, cmd=cmd)
+        time.sleep(10)
+
+        endtime = datetime.datetime.now() + datetime.timedelta(seconds=180)
+        while datetime.datetime.now() < endtime:
+            if self.check_mgr_exists_on_host(host=hostname):
+                log.info(f"Mgr {hostname} is add to the cluster.")
+                return True
+            log.info(
+                "Mgr daemon still not added.sleeping for 5 seconds and checking again"
+            )
+            time.sleep(5)
+        return False
+
+    def remove_mgr_service(self, host) -> bool:
+        """
+        Method to remove the mgr service from the given host
+
+        Args:
+            host: name of the host from where mgr needs to be removed
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        mgr_daemon_list = self.get_mgr_daemon_list()
+        new_mgr_daemon = ""
+        for node in mgr_daemon_list:
+            if host.hostname in node:
+                new_mgr_daemon = node.strip()
+                break
+
+        host_name = host.hostname
+        if not self.check_mgr_exists_on_host(host=host_name):
+            log.info(f"MGR daemon not present on the host {host_name}. Returning Pass")
+            return True
+
+        # setting mgr daemon unmanaged to true
+        unmanaged_cmd_true = " ceph orch apply mgr  --unmanaged=true"
+        self.client.exec_command(sudo=True, cmd=unmanaged_cmd_true)
+
+        cmd = f"ceph orch daemon rm mgr.{new_mgr_daemon} --force"
```

Comment:
Interesting..the official documented way for removal of Manager daemon is by reapplying the manager service with required config/spec
Ref: https://www.ibm.com/docs/en/SSEG27_6/operations/proc_orch_removing-the-manager-daemons-using-the-ceph-orchestrator.html
I wonder which method we should ideally adopt in our scripts
There are two options:
1. `ceph orch daemon rm` -> Needs unmanaged set to true
2. Remove label from the host first, then execute `ceph orch apply mgr --placement='label:mgr'` to reapply mgr service -> should ideally not need unmanaged flag to be set

@pdhiran @SrinivasaBharath opinions?

Reply:
@neha-gangadhar

Reply:
FYI- The following is the document to remove the daemons.
https://docs.ceph.com/en/reef/cephadm/services/#removing-a-daemon-from-a-host-manually